docs/user-guide/migrating-to-2.0.0.md

```diff
@@ -0,0 +1,49 @@
+# Migrating to v2.0.0
+
+ESLint v2.0.0 is the second major version release. As a result, there are some significant changes between how ESLint worked during its life in 0.x and 1.x and how it will work going forward. These changes are the direct result of feedback from the ESLint community of users and were not made without due consideration for the upgrade path. We believe that these changes make ESLint even better, and while some work is necessary to upgrade, we hope the pain of this upgrade is small enough that you will see the benefit of upgrading.
+
+**Important:** If you are upgrading from 0.x, please refer to [Migrating to 1.0.0](./migrating-to-1.0.0) as your starting point.
+
+## Rule Schema Changes
+
+Due to a quirk in the way rule schemas worked, it was possible that you'd need to account for the rule severity (0, 1, or 2) in a rule schema if the options were sufficiently complex. That would result in a schema such as:
+
+```js
+module.exports = {
+    "type": "array",
+    "items": [
+        {
+            "enum": [0, 1, 2]
+        },
+        {
+            "enum": ["always", "never"]
+        }
+    ],
+    "minItems": 1,
+    "maxItems": 2
+};
+```
+
+This was confusing to rule developers as it seemed that rules shouldn't be in charge of validating their own severity. In 2.0.0, rules no longer need to check their own severity.
+
+**To address:** If you are exporting a rule schema that checks severity, you need to make several changes:
+
+1. Remove the severity from the schema
+1. Adjust `minItems` from 1 to 0
```

Comment:
Should this also say "subtract 1", like the line below? It's possible that there might be a minimum required configuration item so that `minItems` would go from (e.g.) 2 to 1.